File: Frontend/src/types/event.ts
export type EventCategory =
  | 'Hotel'
  | 'Restaurant'
  | 'Landmark'
  | 'Travel'
  | 'Activity'
  | 'Shopping'
  | 'Transport'
  | 'Other';

export interface ItineraryEvent {
  id: string;
  version: string;
  itineraryId: string;
  title: string;
  description: string;
  category: EventCategory;
  color: string;
  startDateTime: string;
  endDateTime: string;
  timezone: string;
  location: string;
  locationAddress: string;
  locationLat: number | null;
  locationLng: number | null;
  cost: number;
  createdBy: string;
  updatedBy: string;
  createdAt: string;
  updatedAt: string;
}

export interface EventAuditSnapshot {
  id: string;
  itineraryId: string;
  title: string;
  description: string | null;
  category: EventCategory;
  color: string | null;
  startDateTime: string;
  endDateTime: string;
  timezone: string;
  location: string | null;
  locationAddress: string | null;
  locationLat: number | null;
  locationLng: number | null;
  cost: number | null;
  updatedBy: string;
  updatedAt: string;
}

export interface EventAuditLog {
  id: string;
  eventId: string;
  itineraryId: string;
  action: 'Created' | 'Updated' | 'Deleted';
  summary: string;
  snapshot: EventAuditSnapshot;
  changedBy: string;
  changedAt: string;
}

export interface LocationSuggestion {
  id: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
}

File: Backend/TravelPlannerApp/src/TravelPlannerApp.Domain/Entities/Event.cs
using TravelPlannerApp.Domain.Enums;

namespace TravelPlannerApp.Domain.Entities;

public sealed class Event
{
    public string Id { get; set; } = string.Empty;
    public string ItineraryId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public EventCategory Category { get; set; }
    public string? Color { get; set; }
    public DateTime StartDateTimeLocal { get; set; }
    public DateTime EndDateTimeLocal { get; set; }
    public string Timezone { get; set; } = string.Empty;
    public string? Location { get; set; }
    public string? LocationAddress { get; set; }
    public decimal? LocationLat { get; set; }
    public decimal? LocationLng { get; set; }
    public decimal? Cost { get; set; }
    public string CreatedById { get; set; } = string.Empty;
    public string UpdatedById { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; }
    public DateTime UpdatedAtUtc { get; set; }

    public Itinerary? Itinerary { get; set; }
    public User? CreatedBy { get; set; }
    public User? UpdatedBy { get; set; }
    public ICollection<EventAuditLog> AuditLogs { get; set; } = new List<EventAuditLog>();
}
